opensrp-opd/src/main/java/org/smartregister/opd/presenter/OpdProfileOverviewFragmentPresenter.java

```diff
@@ -102,17 +109,47 @@ private void generateYamlConfigList(@NonNull Facts facts, @NonNull List<YamlConf
-        facts.put("date_of_appointment", checkIn.getAppointmentDueDate());
-
-        if (visit.getVisitDate() != null && checkIn.getAppointmentDueDate() != null) {
-            facts.put("visit_to_appointment_date", getVisitToAppointmentDateDuration(visit.getVisitDate(), checkIn.getAppointmentDueDate()));
+    @Override
+    public void setDataFromCheckIn(@Nullable OpdCheckIn checkIn, @Nullable OpdVisit visit, @Nullable OpdDetails opdDetails, @NonNull Facts facts) {
+        String unknownString = getString(R.string.unknown);
+        if (checkIn != null) {
+            if (client != null && AllConstants.FEMALE_GENDER.equalsIgnoreCase(client.getColumnmaps().get("gender"))) {
+                OpdFactsUtil.putNonNullFact(facts, OpdConstants.FactKey.ProfileOverview.PREGNANCY_STATUS, checkIn.getPregnancyStatus());
+            }
+
+            String currentHivResult = checkIn.getCurrentHivResult() != null ? checkIn.getCurrentHivResult() : unknownString;
+
+
+            OpdFactsUtil.putNonNullFact(facts, OpdConstants.FactKey.ProfileOverview.PATIENT_ON_ART, checkIn.getIsTakingArt());
+
+            // Client is currently checked-in, show the current check-in details
+            if (OpdLibrary.getInstance().isClientCurrentlyCheckedIn(visit, opdDetails)) {
+                OpdFactsUtil.putNonNullFact(facts, OpdConstants.FactKey.ProfileOverview.IS_PREVIOUSLY_TESTED_HIV, checkIn.getHasHivTestPreviously());
+                OpdFactsUtil.putNonNullFact(facts, OpdConstants.FactKey.ProfileOverview.PREVIOUSLY_HIV_STATUS_RESULTS, checkIn.getHivResultsPreviously());
+
+                OpdFactsUtil.putNonNullFact(facts, OpdConstants.FactKey.ProfileOverview.CURRENT_HIV_STATUS, currentHivResult);
+
+                OpdFactsUtil.putNonNullFact(facts, OpdConstants.FactKey.ProfileOverview.VISIT_TYPE, checkIn.getVisitType());
+                OpdFactsUtil.putNonNullFact(facts, OpdConstants.FactKey.ProfileOverview.APPOINTMENT_SCHEDULED_PREVIOUSLY, checkIn.getAppointmentScheduledPreviously());
+                OpdFactsUtil.putNonNullFact(facts, OpdConstants.FactKey.ProfileOverview.DATE_OF_APPOINTMENT, checkIn.getAppointmentDueDate());
+            } else {
+                OpdFactsUtil.putNonNullFact(facts, OpdConstants.FactKey.ProfileOverview.HIV_STATUS, currentHivResult);
+            }
+        } else {
+            if (client != null && unknownString != null) {
+                if (AllConstants.FEMALE_GENDER.equalsIgnoreCase(client.getColumnmaps().get("gender"))) {
+                    OpdFactsUtil.putNonNullFact(facts, OpdConstants.FactKey.ProfileOverview.PREGNANCY_STATUS, unknownString);
+                } else {
+                    OpdFactsUtil.putNonNullFact(facts, OpdConstants.FactKey.ProfileOverview.HIV_STATUS, unknownString);
+                }
+            }
+        }
+
+        boolean shouldCheckIn = OpdLibrary.getInstance().canPatientCheckInInsteadOfDiagnoseAndTreat(visit, opdDetails);
```

Comment:
I would have to negate the result of `isClientCurrentlyCheckedIn`

1. I found that the simple distinction in method names make it easy to read the code instead of trying to read negated method results
2. `isClientCurrentlyCheckedIn` means something completely different to `canPatientCheckInInsteadOfDiagnoseAndTreat`. Currently in my mind and as I wrote it, it's more of a case where they share code but might not always mean the same in cases where the patient is treated but going to the profile still enables you to checkin the patient.

What are you thoughts? Something simpler maybe?

Reply:
Whichever way you choose... i will be okay with it